### PR TITLE
Update markdownlinting packages and configs

### PR DESCRIPTION
**Pull request by khiga8**
### Description

- Introduce `markdownlint-github` and configure rules.
- Swap `markdownlint-cli` to `markdownlint-cli2` which is **compatible** with the [markdownlint VSCode extension](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint). With `markdownlint-cli`, the extension won't respect the configs that are set.
- Add [markdownlint VSCode extension](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) to `.devcontainer.json` so every new codespace comes with it installed.

### Integration

> Does this change require any updates to code in production?

No

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews

### CODE REVIEW

**Comment by khiga8** [comment]
This PR will introduce the `no-generic-link-text` but I realized that it will falsely flag references to the `[Link]` component. `markdownlint` does support disable comments but due to how markdown files are generated in PVC, this doesn't seem like a viable path since the disable would just get overwritten.

Maybe we can have all component links have component as part of the name like `Read more at [Blankslate component]` or `Learn more at [Link component]`... but also I'm not really sure about this... I don't want to make an executive decision on this now.

The option I'm leaning towards is to just let exceptions be defined for the rule in the markdown config, so `link` will be specified as an exception. I opened a PR here which I'd appreciate a review on: [Staff-only: markdownlint-github PR](https://github.com/github/markdownlint-github/pull/27).

Let me know if you have additional thoughts: @primer/rails-reviewers.

File: .markdownlint-cli2.cjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const options = require('@github/markdownlint-github').init({
+  "line-length": false,
+  "ol-prefix": false,
+  "no-trailing-punctuation": false,
+  "no-inline-html": false,
+  "first-line-heading": false,
+  // Rules that we want to enable
+  "no-duplicate-heading": false
```
Comment (khiga8) [inline .markdownlint-cli2.cjs:8]: I think we'll want to eventually enable this `no-duplicate-heading` rule. 

With Primer docs, anchor links are generated from the headings. When headings are duplicated on a page, the generated anchor links will always go to the first instance of the heading. 

One example is [#autocorrectable heading](https://primer.style/view-components/linting#autocorrectable) which appears twice on different sections of the page, and share the same anchor link.

**Comment by jonrohan** [comment]
Nice! We have our lint system setup so it won't run unless there's a change to a markdown file. You should test this new config by adding an error somewhere.

**Comment by khiga8** [comment]
> Nice! We have our lint system setup so it won't run unless there's a change to a markdown file. You should test this new config by adding an error somewhere.

Here's [failed run](https://github.com/primer/view_components/actions/runs/3788886862/jobs/6442115543) after introducing an error.

Does https://github.com/primer/view_components/pull/1732#issuecomment-1366020678 sound reasonable to you?

**Review by jonrohan — APPROVED** [review]
> Does [#1732 (comment)](https://github.com/primer/view_components/pull/1732#issuecomment-1366020678) sound reasonable to you?

Yep sounds good to me 👍🏻

**Comment by khiga8** [comment]
I am not authorized to merge. This is ready! Please merge freely.

**Review by jonrohan — APPROVED** [review]

**Pull request merged**